SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -125,6 +125,13 @@ FSpatialLoadBalancingHandler::EvaluateActorResult FSpatialLoadBalancingHandler::
 					UE_LOG(LogSpatialLoadBalancingHandler, Error,
 						   TEXT("Load Balancing Strategy returned invalid virtual worker for actor %s"), *Actor->GetName());
 				}
+				else if (!bShouldHaveAuthority && NewAuthVirtualWorkerId == NetDriver->LoadBalanceStrategy->GetLocalVirtualWorkerId())
```

Comment:
Interesting, is it something that happens with our existing load balancing strategies?
While hardening this bit of code is good, we should also fix the load balancing strategy that allows this situation to arise.

Reply:
Not that I'm aware - I'll clarify also in PR description that this is just to guard against bad custom LBS code where ShouldHaveAuthority and WhoShouldHaveAuthority is inconsistent.

Reply:
Is it worth adding a check inside WhoShouldHaveAuthority when returning a worker ID that if it's the local worker then ShouldHaveAuthority also confirms that?